internal/async_helpers.go

```diff
@@ -35,6 +35,10 @@ func (c *Client) GetJobID(
 	}
 	resp.Body.Close()
 
+	if resp.StatusCode != http.StatusAccepted {
```

Comment:
To be more safe, can we check instead if it's above a 3xx range? That sounds safer

Also you need to move this up before we read the body, as it's kind of useless at this check

Reply:
I'll change to cover the range instead.

But when it comes to checking code before reading the body I disagree. Status code above 3xx does not mean we do not get the body. With this approach if we get error from the API, then error message will contain the underlying issue. Without the body we can't inform user what is the cause of the issue.

Reply:
Yes, but you're not returning the error msg from the body in the response

Reply:
Oh sorry I see you are, but that's not the right way to do it

Reply:
It should be a structured error, not raw bytes

Reply:
Do you have an example of a structured error in the SDK? I mostly see values `%v` added into errors.

Plus in my case error message is not raw bytes, it is turned into string with `%s`, making it human readable.
Would `fmt.Errorf("error with status code %s: %s", httpResp.Status, respBody)` as in https://github.com/mslmio/oxylabs-sdk-go/blob/master/ecommerce/resp.go#L607 be fine?

Reply:
Don't have an example, depends on if it's even documented / structured in the docs in the first place.

We could just merge this then for now, though then do include the HTTP code like the example you linked

Reply:
Structuring errors returned by API would be good, but out of the scope of this fix.